File: btypes_big_integer/src/main/clojure/btypes/src/btypes/butils.clj
(ns btypes.butils
  (:require [clojure.set :as s])
  (:require [btypes.bboolean])
  (:require [btypes.bset])
  (:import (de.hhu.stups.btypes BBoolean
             BSet))
  (:gen-class :name de.hhu.stups.btypes.BUtils))


(gen-class 
     :name de.hhu.stups.btypes.BUtils
     :methods [^:static [BOOL [] de.hhu.stups.btypes.BSet]])

(def BOOL_CONST (delay (de.hhu.stups.btypes.BSet. #{(de.hhu.stups.btypes.BBoolean. true) (de.hhu.stups.btypes.BBoolean. false)})))


(defn BOOL []
  @BOOL_CONST)
  

File: btypes_big_integer/src/main/js/btypes_stups/BUtils.ts
import {BSet} from "https://www3.hhu.de/stups/models/visb/btypes/BSet.js";
import {BBoolean} from "https://www3.hhu.de/stups/models/visb/btypes/BBoolean.js";

export class BUtils {
    static readonly BOOL: BSet<BBoolean>  = new BSet<BBoolean>(new BBoolean(true), new BBoolean(false));
}

export class SelectError extends Error {
    constructor(msg: string) {
        super(msg);
        Object.setPrototypeOf(this, SelectError.prototype);
    }
}

export class PreconditionOrAssertionViolation extends Error {
    constructor(msg: string) {
        super(msg);
        Object.setPrototypeOf(this, PreconditionOrAssertionViolation.prototype);
    }
}

export class VariantViolation extends Error {
    constructor(msg: string) {
        super(msg);
        Object.setPrototypeOf(this, SelectError.prototype);
    }
}

export class LoopInvariantViolation extends Error {
    constructor(msg: string) {
        super(msg);
        Object.setPrototypeOf(this, SelectError.prototype);
    }
}